kobo/apps/hook/tests/test_api_hook.py

```diff
@@ -0,0 +1,190 @@
+
+        v_uid = self.asset.latest_deployed_version.uid
+        submission = {
+            "__version__": v_uid,
+            "q1": u"¿Qué tal?",
+            "id": 1
+        }
+        self.asset.deployment._mock_submission(submission)
+        self.asset.save(create_version=False)
+        settings.CELERY_ALWAYS_EAGER = True
+
+    def _create_hook(self):
+        url = reverse("hook-list", kwargs={"parent_lookup_asset": self.asset.uid})
+        data = {
+            "name": "some external service with token",
+            "endpoint": "http://external.service.local/",
+            "settings": {
+                "custom_headers": {
+                    "X-Token": "1234abcd"
+                }
+            }
+        }
+        response = self.client.post(url, data, format=INSTANCE_FORMAT_TYPE_JSON)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED,
+                         msg=response.data)
+        hook = self.asset.hooks.last()
+        self.assertTrue(hook.active)
+        return hook
+
+    def test_anonymous_access(self):
+        hook = self._create_hook()
+        self.client.logout()
+
+        list_url = reverse("hook-list", kwargs={
+            "parent_lookup_asset": self.asset.uid
+        })
+
+        response = self.client.get(list_url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        detail_url = reverse("hook-detail", kwargs={
+            "parent_lookup_asset": self.asset.uid,
+            "uid": hook.uid,
+        })
+
+        response = self.client.get(detail_url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        log_list_url = reverse("hook-log-list", kwargs={
+            "parent_lookup_asset": self.asset.uid,
+            "parent_lookup_hook": hook.uid,
+        })
+
+        response = self.client.get(log_list_url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+    def test_not_owner_access(self):
+        hook = self._create_hook()
+        self.client.logout()
+        self.client.login(username="anotheruser", password="anotheruser")
+
+        list_url = reverse("hook-list", kwargs={
+            "parent_lookup_asset": self.asset.uid
+        })
+
+        response = self.client.get(list_url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+        detail_url = reverse("hook-detail", kwargs={
+            "parent_lookup_asset": self.asset.uid,
+            "uid": hook.uid,
+        })
+
+        response = self.client.get(detail_url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+        log_list_url = reverse("hook-log-list", kwargs={
+            "parent_lookup_asset": self.asset.uid,
+            "parent_lookup_hook": hook.uid,
+        })
+
+        response = self.client.get(log_list_url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+
+    def test_create_hook(self):
```

Comment:
Since I was talking about `AssetOwnerFilterBackend` earlier, I got to wondering: is there a test of trying to create a hook on an asset that the user does not own?

(not blocking)

Reply:
No there are no tests for that case. I will add it in `rest_email_notification` to make the merge easier.